cmake/modules/packages/FindPCRE2.cmake

```diff
@@ -9,20 +9,20 @@
 # Find the native PCRE2 headers and libraries.
 
 find_path(PCRE2_INCLUDE_DIR NAMES pcre2.h)
-find_library(PCRE2_LIBRARY NAMES pcre2-8 pcre2-8d pcre2-8-static pcre2-8-staticd NAMES_PER_DIR)
+find_library(PCRE2-8_LIBRARY NAMES pcre2-8 pcre2-8d pcre2-8-static pcre2-8-staticd NAMES_PER_DIR)
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(PCRE2
-                                  FOUND_VAR PCRE2_FOUND
-                                  REQUIRED_VARS PCRE2_LIBRARY PCRE2_INCLUDE_DIR)
-mark_as_advanced(PCRE2_INCLUDE_DIR PCRE2_LIBRARY)
-if(PCRE2_FOUND)
-    set(PCRE2_LIBRARIES ${PCRE2_LIBRARY})
+find_package_handle_standard_args(PCRE2-8
+                                  FOUND_VAR PCRE2-8_FOUND
+                                  REQUIRED_VARS PCRE2-8_LIBRARY PCRE2_INCLUDE_DIR)
```

Comment:
We probably still need a
~~~cmake
set(PCRE2_FOUND "${PCRE2-8_FOUND}")
~~~

Reply:
Do we need `PCRE2-8_FOUND`? Or can we replace it as `FOUND_VAR PCRE2_FOUND`?

Reply:
No, we don't need `PCRE2-8_FOUND`. I'm just not so familiar with the options of `find_package_handle_standard_args`.

Reply:
Research shows:
- `FOUND_VARS` is deprecated since cmake 3.3
- ci complains about `find_package_handle_standard_args` being used with a name different from the current Find module. Latest commit should fix that.